javascript/javascript2/week2/main.js

```diff
@@ -0,0 +1,83 @@
+console.log('Script loaded');
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector('section.products ul');
+console.log(productsUl);
+
+//Displaying products in productsUl
+function renderProducts(products) {
+    let productLi, productName, productPrice, productRating;
+    products.forEach(product => {
+        productLi = document.createElement('li');
+        productName = document.createElement('h2');
+        productPrice = document.createElement('p');
+        productRating = document.createElement('p');
+
+        productName.innerHTML = product.name;
+        productPrice.innerHTML = `Price: ${product.price}`;
+        productRating.innerHTML = `Rating: ${product.rating}`;
+
+
+        productLi.appendChild(productName);
+        productLi.appendChild(productPrice);
+        productLi.appendChild(productRating)
+        productsUl.appendChild(productLi);
+    });
+}
+renderProducts(products);
+
+
+//Searching a product 
+const searchBox = document.querySelector('.search input');
+searchBox.addEventListener('keyup', myFunction);
+function myFunction() {
+    const searchBoxValue = searchBox.value.toLowerCase();
+    const filteredArray = products.filter(product => {
+        if (product.name.toLowerCase().includes(searchBoxValue)) {
+            productsUl.innerHTML = '';
+            return product.name;
+        }
```

Comment:
This is only a suggestion. An alternative is to return early.
```suggestion
        if (!product.name.toLowerCase().includes(searchBoxValue)) {
            return;
        }
        
        productsUl.innerHTML = '';
        return product.name;
```

Reply:
Thanks Cameron. did as you mentioned.